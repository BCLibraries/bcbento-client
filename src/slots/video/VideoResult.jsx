import React from 'react';
import OnlineVideo from './OnlineVideo';
import PhysicalVideo from './PhysicalVideo';

function VideoResult({item}) {
    const video = item.isElectronic ? <OnlineVideo video={item}/> : <PhysicalVideo video={item} />;

    return <li className="catalog-result-item col-md-4">
        {video}
    </li>
}

export default VideoResult;