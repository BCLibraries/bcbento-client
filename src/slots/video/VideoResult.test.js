import renderer from 'react-test-renderer';
import VideoResult from './VideoResult.jsx';

it('renders correctly', () => {
    const item = {
        "id": "ALMA-BC51425736950001021",
        "title": "Peru Globe Trekker",
        "date": "[2012], c1998",
        "type": "Video",
        "creator": null,
        "contributors": [
            "555 Productions, Inc",
            "Films for the Humanities & Sciences (Firm)",
            "Films Media Group"
        ],
        "coverImages": [
            {
                "url": "no_cover"
            }
        ],
        "available": null,
        "isPhysical": false,
        "isElectronic": true,
        "linkableId": "ALMA-BC51425736950001021",
        "screenCap": "https://fod.infobase.com/image/50468",
        "format": "1 streaming video file (48 min.) : sd., col.",
        "mms": "99134178990001021",
        "availability": null
    };

    const component = renderer.create(
        <VideoResult item={item}/>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
