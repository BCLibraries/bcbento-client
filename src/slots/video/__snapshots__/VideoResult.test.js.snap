// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<li
  className="catalog-result-item col-md-4"
>
  <div
    className="online-video"
  >
    <a
      href="https://mlib.bc.edu/reserves-api/items/99134178990001021"
    >
      <div
        className="online_video__thumb"
      >
        <img
          alt="Watch Peru Globe Trekker online"
          className="online-video-thumb__screenshot"
          onError={[Function]}
          src="https://fod.infobase.com/image/50468"
        />
      </div>
    </a>
    <div>
      <div
        className="catalog-result-item__type"
      >
        Video
      </div>
      <h3
        className="online-video__media-heading media-heading"
      >
        <a
          className="online-video__title"
          href="https://bc-primo.hosted.exlibrisgroup.com/primo-explore/fulldisplay?docid=ALMA-BC51425736950001021&context=L&tab=bcl_only&search_scope=bcl&vid=bclib_new&lang=en_US"
        >
          Peru Globe Trekker
        </a>
      </h3>
    </div>
    [2012], c1998
    <div
      className="online-video__creator"
    >
      555 Productions, Inc
    </div>
    <div
      className="online-video__watch-link"
    >
      <a
        href="https://mlib.bc.edu/reserves-api/items/99134178990001021"
      >
        Watch online
      </a>
    </div>
  </div>
</li>
`;
